lib/lacerda/compare/json_schema.rb

```diff
@@ -189,24 +183,37 @@ def schema_contains?(options)
       private
 
       def properties_contained?
-        @contained_schema['properties'].each do |property, contained_property|
-          resolved_contained_property = data_for_pointer(contained_property, @contained_schema)
-          containing_property = @containing_schema['properties'][property]
-          if !containing_property
-            _e(:ERR_MISSING_DEFINITION, [@initial_location, property], "(in publish.mson)")
-          else
-            resolved_containing_property = data_for_pointer(
-              containing_property,
-              @containing_schema
-            )
-            schema_contains?(
-              publish: resolved_containing_property,
-              consume: resolved_contained_property,
-              location: [property]
-            )
-          end
+        # success is used to ensure we have no errors.
+        success = @contained_schema['properties'].map do |name, content|
+          property_contained?(name, content)
+        end.all? {|is_property_contained| is_property_contained }
+        success && @errors.empty?
+      end
+
+      def property_contained?(property_name, content)
+        resolved_contained_property = data_for_pointer(content, @contained_schema)
+        containing_property = @containing_schema['properties'][property_name]
+
+        if !containing_property
+          return _e(:ERR_MISSING_DEFINITION, [@initial_location, property_name], "(in publish.mson)")
         end
-        @errors.empty?
+
+        # Make sure required properties in consume are required in publish

```

Comment:
Split the method, and added the `required` check here too. This makes [this new spec](https://github.com/moviepilot/lacerda/pull/43/files#diff-52c44d514913a06380abba99fcf86c7cR364) pass, because before we just looked the `required` inside definitions